packages/flutter/lib/src/material/search.dart

```diff
@@ -48,6 +48,9 @@ import 'theme.dart';
 /// route created by this method is pushed to the nearest navigator to the
 /// given `context`. It can not be `null`.
 ///
+/// The `maintainState` argument is used to determine if the route should remain
+/// in memory when it is inactive. By default, `maintainState` is `false`.
```

Comment:
Shall we refer to the original doc for more information? 
```suggestion
/// The `maintainState` argument is used to determine if the route should remain
/// in memory when it is inactive (see [ModalRoute.maintainState] for more details].
/// By default, `maintainState` is `false`.
```

Also, do you think it's possible to provide examples on how this might be useful?